src/sbvr-api/permissions.ts

```diff
@@ -684,6 +685,200 @@ const getAlias = (name: string) => {
+					// this relations ship means that:
+					//   arr[0] is the local field
+					//   arr[1] is the reference to the other resource, that joins this resource
+					//   arr[1][0] is the name of the other resource (actor in the example)
+					//   arr[1][1] is the name of the field on the other resource
+					//
+					// this therefore defines that the local field `actor` needs
+					// to match the `id` of the `actor` resources for the join
+
+					const possibleTargetResourceName = arr[1][0];
+
+					// Skip this if we already shortcut this connection
+					if (possibleTargetResourceName.endsWith('$bypass')) {
+						return;
+					}
+
+					const targetResource =
+						abstractSqlModel.tables[possibleTargetResourceName];
+					const targetResourceEscaped = sqlNameToODataName(
+						targetResource?.name ?? possibleTargetResourceName,
+					);
+
+					// This is either a translated or bypassed resource we don't
+					// mess with these
+					if (targetResourceEscaped.includes('$')) {
+						return;
+					}
+
+					let foundCanAccessLink = false;
+
+					try {
+						const odata = memoizedParseOdata(`/${targetResourceEscaped}`);
+
+						const collapsedPermissionFilters = buildODataPermission(
+							permissionsLookup,
+							methodPermissions.GET,
+							vocabulary,
+							targetResourceEscaped,
+							odata,
+						);
+
+						_.set(
+							odata,
+							['tree', 'options', '$filter'],
+							collapsedPermissionFilters,
+						);
+
+						const canAccessFunction: ResourceFunction = function(
+							property: AnyObject,
+						) {
+							// remove method property so that we won't loop back here again at this point
+							delete property.method;
+
+							if (!this.defaultResource) {
+								throw new Error(`No resource selected in AST.`);
+							}
+
+							const targetResourceAST = this.NavigateResources(
+								this.defaultResource,
+								property.name,
+							);
+
+							const targetResourceName = sqlNameToODataName(
+								targetResourceAST.resource.name,
+							);
+							const currentResourceName = sqlNameToODataName(
+								this.defaultResource.name,
+							);
+
+							if (
+								currentResourceName === targetResourceEscaped &&
+								targetResourceName === escapedName
+							) {
+								foundCanAccessLink = true;
+							}
+							// return a true expression to not select the relationship, which might be virtual
+							// this should be a boolean expression, but needs to be a subquery in case it
+							// is wrapped in an `or` or `and`
+							return ['Equals', ['Boolean', true], ['Boolean', true]];
+						};
+
+						// We need execute the
+						const odata2AbstractSQL = new OData2AbstractSQL(
+							originalAbstractSQLModel,
+							{
+								canAccess: canAccessFunction,
+							},
+						);
+
+						try {
+							odata2AbstractSQL.match(
+								odata.tree,
+								'GET',
+								[],
+								odata.binds.length,
+							);
+						} catch (e) {
+							throw new ODataParser.SyntaxError(e);
```

Comment:
Isn't this what is thrown anyway? What other types are you seeing here that you wrap?

Reply:
I need to catch this `Error` https://github.com/balena-io-modules/odata-to-abstract-sql/blob/master/src/odata-to-abstract-sql.ts#L1374

Reply:
Can you give some examples of resources that are failing to resolve? It'd be interesting to see which ones to understand why

Reply:
One resource that is failing there is `application$resin$bypass`, it happens soon when you run the api tests.